### PR TITLE
Add auxv detection

### PR DESCRIPTION
**Pull request by mrda**
Add detect-auxv to help in determining ppc machines

### CODE REVIEW

File: hardware/detect.py [hunk 1]
```diff
@@ -861,6 +862,25 @@ def detect_rtc_clock(hw_lst):
     hw_lst.append(('system', 'rtc', 'utc', rtc.get_rtc()))
 
 
+def detect_auxv(hw_lst):
+    new_env = os.environ.copy()
+    new_env["LD_SHOW_AUXV"] = "1"
+
+    uuid_cmd = Popen("/bin/true",
```
Comment (tbreeds) [inline hardware/detect.py]: `uuid_cmd` seems strange here. `auxv_cmd` or just `cmd` perhaps?

File: hardware/detect.py [hunk 2]
```diff
@@ -861,6 +862,25 @@ def detect_rtc_clock(hw_lst):
...
+                     env=new_env,
+                     stdout=PIPE)
+    stdout, err = uuid_cmd.communicate()
+    if err is not None:
```
Comment (tbreeds) [inline hardware/detect.py]: Can we have something printed to stderr as a breadcrumb

File: hardware/detect.py [hunk 3]
```diff
@@ -51,8 +51,9 @@
 from hardware import rtc
 from hardware import smart_utils
 
-
 SIOCGIFNETMASK = 0x891b
+AUXV_FLAGS = ("AT_HWCAP", "AT_HWCAP2", "AT_PAGESZ",
+              "AT_FLAGS", "AT_PLATFORM")
```
Comment (tbreeds) [inline hardware/detect.py]: I'd like to see AT_BASE_PLATFORM exposed/captured here.  It's meaningful on powerpc:
```
# LD_SHOW_AUXV=1 /bin/true
AT_DCACHEBSIZE:  0x80
AT_ICACHEBSIZE:  0x80
AT_UCACHEBSIZE:  0x0
AT_SYSINFO_EHDR: 0x3fff95170000
AT_HWCAP:        true_le archpmu vsx arch_2_06 dfp ic_snoop smt mmu fpu altivec ppc64 ppc32
AT_PAGESZ:       65536
AT_CLKTCK:       100
AT_PHDR:         0x10000040
AT_PHENT:        56
AT_PHNUM:        9
AT_BASE:         0x3fff95190000
AT_FLAGS:        0x0
AT_ENTRY:        0x1000147c
AT_UID:          0
AT_EUID:         0
AT_GID:          0
AT_EGID:         0
AT_SECURE:       0
AT_RANDOM:       0x3ffff55c9492
AT_HWCAP2:       htm-nosc vcrypto tar isel ebb dscr htm arch_2_07
AT_EXECFN:       /bin/true
AT_PLATFORM:     power8
AT_BASE_PLATFORM:power8
# 
```

However adding that here will mean a tweak to the test as we'd no longer get all the keys in `detect.AUXV_FLAGS` in the hw_list

File: hardware/detect.py (hunk 1 above)
Comment (mrda) [reply hardware/detect.py]: Done.

File: hardware/detect.py (hunk 3 above)
Comment (mrda) [reply hardware/detect.py]: I've done that by introducing AUXV_OPT_FLAGS which may or not be present on a particular architecture.

File: hardware/detect.py (hunk 2 above)
Comment (mrda) [reply hardware/detect.py]: Done

File: hardware/tests/test_detect.py [hunk 4]
```diff
@@ -227,6 +228,114 @@ def test_clean_tuples(self):
                                   b'h\xc3\xa9llo', 1)]),
             [(u'\ufffd' * 4, u'\ufffd' * 4, u'h\xe9llo', 1)])
 
+    @mock.patch.object(detect, 'Popen')
+    @mock.patch('os.environ.copy')
+    def test_detect_auxv_x86_succeed(self, mock_environ_copy, mock_popen):
+        test_data = {
```
Comment (elfosardo) [inline hardware/tests/test_detect.py]: can you please move all the test_data outside of this file?
like we do for other tests, can be a file in samples

File: hardware/tests/test_detect.py [hunk 5]
```diff
@@ -227,6 +228,114 @@ def test_clean_tuples(self):
...
+            'AT_HWCAP': ('hwcap', 'bfebfbff'),
+            'AT_HWCAP2': ('hwcap2', '0x0'),
+            'AT_PAGESZ': ('pagesz', '4096'),
+            'AT_FLAGS': ('flags', '0x0'),
+            'AT_PLATFORM': ('platform', 'x86_64'),
+        }
+
+        test_x86_data = textwrap.dedent(
+            """AT_SYSINFO_EHDR:      0x7fff623e0000
```
Comment (elfosardo) [inline hardware/tests/test_detect.py]: ditto
this is basically a huge string, it can be moved outside

File: hardware/tests/test_detect.py [hunk 6]
```diff
@@ -227,6 +228,114 @@ def test_clean_tuples(self):
...
+
+        # NOTE(mrda): x86 doesn't have AUXV_OPT_FLAGS
+        for k in detect.AUXV_FLAGS:
+            t = ('hw', 'auxv', test_data[k][0], test_data[k][1])
+            self.assertTrue(t in hw)
+
+    @mock.patch.object(detect, 'Popen')
+    @mock.patch('os.environ.copy')
+    def test_detect_auxv_ppc8_succeed(self, mock_environ_copy, mock_popen):
+        test_data = {
```
Comment (elfosardo) [inline hardware/tests/test_detect.py]: ditto

File: hardware/tests/test_detect.py [hunk 7]
```diff
@@ -227,6 +228,114 @@ def test_clean_tuples(self):
...
+            'AT_HWCAP': ('hwcap', 'true_le archpmu vsx arch_2_06 dfp ic_snoop smt mmu fpu altivec ppc64 ppc32'),
+            'AT_HWCAP2': ('hwcap2', 'htm-nosc vcrypto tar isel ebb dscr htm arch_2_07'),
+            'AT_PAGESZ': ('pagesz', '65536'),
+            'AT_FLAGS': ('flags', '0x0'),
+            'AT_PLATFORM': ('platform', 'power8'),
+            'AT_BASE_PLATFORM': ('base_platform', 'power8'),
+        }
+
+        test_ppc8_data = textwrap.dedent(
+            """AT_DCACHEBSIZE:  0x80
```
Comment (elfosardo) [inline hardware/tests/test_detect.py]: ditto

File: hardware/tests/test_detect.py (hunk 4 above)
Comment (mrda) [reply hardware/tests/test_detect.py]: Good idea - done

File: hardware/tests/test_detect.py (hunk 6 above)
Comment (mrda) [reply hardware/tests/test_detect.py]: done

File: hardware/tests/test_detect.py (hunk 7 above)
Comment (mrda) [reply hardware/tests/test_detect.py]: done

**Pull request merged**